server: NPE in (*Server).SetupReportingURLs

eb04bb87995beca674264fd22352441c78928d3a

```
16:17 $ ./cockroach start
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4168da5]

goroutine 1 [running]:
panic(0x508d6c0, 0xc8200140c0)
	/usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/server.(*Server).SetupReportingURLs(0x0, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/server/updates.go:74 +0x155
github.com/cockroachdb/cockroach/cli.runStart(0x5d12240, 0x5d476d0, 0x0, 0x0, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/cli/start.go:151 +0x961
github.com/spf13/cobra.(*Command).execute(0x5d12240, 0x5d476d0, 0x0, 0x0, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/spf13/cobra/command.go:565 +0x62c
github.com/spf13/cobra.(*Command).ExecuteC(0x5d0ee40, 0x5d12240, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/spf13/cobra/command.go:656 +0x55c
github.com/spf13/cobra.(*Command).Execute(0x5d0ee40, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/spf13/cobra/command.go:615 +0x2d
github.com/cockroachdb/cockroach/cli.Run(0xc82007e070, 0x1, 0x1, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/cli/cli.go:81 +0x7c
main.main()
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/main.go:37 +0xe6
```

--- a/cli/start.go
+++ b/cli/start.go
@@ -146,14 +146,13 @@ func runStart(_ *cobra.Command, _ []string) error {
 
 	log.Info("starting cockroach node")
 	s, err := server.NewServer(&cliContext.Context, stopper)
+	if err != nil {
+		return fmt.Errorf("failed to start Cockroach server: %s", err)
+	}
 
 	// We don't do this in NewServer since we don't want it in tests.
 	if err := s.SetupReportingURLs(); err != nil {
 		return err
-	}
-
-	if err != nil {
-		return fmt.Errorf("failed to start Cockroach server: %s", err)
 	}
 
 	if err := s.Start(); err != nil {

